src/jit/lsra.cpp

```diff
@@ -5210,17 +5299,14 @@ void LinearScan::allocateRegisters()
             if (currentLocation > prevLocation)
             {
                 freeRegisters(regsToFree);
-                if ((currentLocation > (prevLocation + 1)) && (delayRegsToFree != RBM_NONE))
+                if ((currentLocation > (prevLocation + 1)) && (regsToFree.current != RBM_NONE))
                 {
-                    // We should never see a delayReg that is delayed until a Location that has no RefPosition
+                    // If we have current regsToFree at this point, it means that we had delayRegs.
```

Comment:
Is it necessary to move `delayRegs` to `current` in `freeRegisters`? If we did not do that then we could check `regsToFree.delayRegs` here and won't need this comment.

Ah, then we will need an additional arg for `freeRegisters`. Nevermind then.